### PR TITLE
vcf_slice: Fix basic checklist items for DX apps

### PR DESCRIPTION
**Pull request by msimbirsky**
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [x] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [x] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

#src/code.sh src/code.py
- [x] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [x] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [x] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

### CODE REVIEW

**Comment by msimbirsky** [comment]
Notes (from Eric?)
- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note that GNU parallel should be in an asset bundle (TODO)
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

**Comment by msimbirsky** [comment]
Pull request ready for review

**Review by etal — APPROVED** [review]

File: sequencing/vcf_slice/dxapp.json [hunk 1]
```diff
@@ -1,69 +1,90 @@
 {
   "name": "vcf_slice",
   "title": "Slice VCF File(s)",
-  "summary": "vcf_slice",
+  "summary": "Return a small section of a VCF file or files (similar to tabix).",
   "dxapi": "1.0.0",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "version": "0.1.0", -> "version": "1.0.0",

File: sequencing/vcf_slice/dxapp.json [hunk 2]
```diff
@@ -1,69 +1,90 @@
...
   "categories": [
     "Variation Calling"
   ],
   "inputSpec": [
     {
       "name": "vcf_fn",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "vcf_fn" -> "variants_vcfgzs",
Also need to update accordingly in the app script.

File: sequencing/vcf_slice/dxapp.json [hunk 3]
```diff
@@ -1,69 +1,90 @@
...
+      "help": "Variant files to be sliced, in VCF format.",
       "class": "array:file",
       "optional": false,
-      "patterns" : ["*.vcf.gz"]
+      "patterns": ["*.vcf.gz"]
     },
     {
       "name": "vcfidx_fn",
-      "label": "VCF Index Files",
+      "label": "VCF index files",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "vcfidx_fn" -> "variants_vcfgztbis",
If input is a \*.vcf.gz file, then the index filename should be \*.vcf.tbi.  (Likely the case)
If input is a \*.vcf file, then the index filename should be \*.vcf.idx.

Also need to update variables accordingly in the app script.

File: sequencing/vcf_slice/dxapp.json [hunk 4]
```diff
@@ -1,69 +1,90 @@
...
+      "name": "suffix",
+      "label": "Output file name suffix (optional)",
+      "help": "(Optional) Include this given string in the output VCF filename extension, e.g. '*.sliced.SUFFIX.vcf.gz'.",
       "class": "string",
       "optional": true
     }
   ],
   "outputSpec": [
     {
       "name": "vcf_out",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "vcf_out" -> "out_variants_vcfgzs"
The variable name seems to be misplaced with "out_variants_vcfgztbis" (below)

Also need to update variables accordingly in the app script.

File: sequencing/vcf_slice/dxapp.json [hunk 5]
```diff
@@ -1,69 +1,90 @@
...
-      "label": "Sliced VCF Files",
-      "class": "array:file"
+      "label": "Sliced VCF files",
+      "help": "Variant file containing only target regions, in VCF format.",
+      "class": "array:file",
+      "patterns": ["*.vcf.gz.tbi"]
     },
     {
       "name": "vcfidx_out",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "vcfidx_out" -> "out_variants_vcfgztbis"
The variable name seems to be misplaced with "out_variants_vcfgzs" (above)

Also need to update variables accordingly in the app script.

File: sequencing/vcf_slice/dxapp.json [hunk 6]
```diff
@@ -1,69 +1,90 @@
...
+        "*": { "instanceType": "mem1_ssd1_x4" }
+      },
+      "assetDepends": [{"id": "record-FGpf7zQ0kfbZxj6q4ZzQ2BPG"}]
+    },
+    "azure:westus": {
+      "systemRequirements": {
+        "*": { "instanceType": "azure:mem2_ssd1_x4" }
+      },
+      "assetDepends": [{"id": "record-FGpfZ1j95qzXV43bP5y5JKgV"}]
+    }
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: Please update record IDs to:
aws:us-east-1: record-FJ3zf8j08zVZFzqYG88Pbk7z
azure:westus: record-FJ4058j9Vf9ZBBVB2Q7b3VQz

File: sequencing/vcf_slice/dxapp.json [hunk 7]
```diff
@@ -1,69 +1,90 @@
...
   },
-  "resources" : "project-BYpFk1Q0pB0xzQY8ZxgJFv1V",
-  "openSource" : true,
-  "developers" : ["user-tnperson", "user-johnwallace123", "user-auv13", "user-atf3"],
-  "authorizedUsers": ["org-geisinger_affiliate", "org-geisinger", "org-support", "user-codushlaine", "user-semanti.mukherjee", "user-alexander.li", "user-shannon.bruse", "user-jan.freudenberg", "user-crisvh"]
+  "details": {
+        "upstreamVersion": "",
+        "whatsNew": "",
```
Comment (yihchii) [inline sequencing/vcf_slice/dxapp.json]: "whatsNew": "* 1.0.0 First version of the Slice VCF File(s) app."

**Review by yihchii — COMMENTED** [review]
Some comments:

1. Please update the app version to 1.0.0.
2. Please follow the variable nomenclature.
3. There seems to have two output variables naming mixed. Please confirm and fix.
4. Please update to the latest asset record IDs.

File: sequencing/vcf_slice/Readme.md [hunk 8]
```diff
@@ -1,21 +1,32 @@
-<!-- dx-header -->
 # Slice VCF File(s) (DNAnexus Platform App)
 
-Return a small section of a VCF file (similar to tabix).  For large output, many small regions, or subsetting samples, use subset_vcf instead.
+Return a small section of a VCF file (similar to tabix).  For large output,
```
Comment (slzarate) [inline sequencing/vcf_slice/Readme.md:3]: Make into 1 line

File: sequencing/vcf_slice/Readme.md [hunk 9]
```diff
@@ -1,21 +1,32 @@
...
+Return a small section of a VCF file (similar to tabix).  For large output, 
+many small regions, or subsetting samples, use the app `select_variants` instead.
 
-This is the source code for an app that runs on the DNAnexus Platform.
-For more information about how to run or modify it, see
-https://wiki.dnanexus.com/.
-<!-- /dx-header -->
+## App Inputs
 
+This app requires 1 or more variant files in VCF format to be filtered to the
```
Comment (slzarate) [inline sequencing/vcf_slice/Readme.md]: Make into 1 line, also specify that it's (`*.vcf.gz`)

File: sequencing/vcf_slice/Readme.md [hunk 10]
```diff
@@ -1,21 +1,32 @@
...
+target region and a corresponding Tabix index file as well as a region file
```
Comment (slzarate) [inline sequencing/vcf_slice/Readme.md]: Specify that it's (`*.vcf.gz.tbi`) and whatever the format for the region file is

File: sequencing/vcf_slice/Readme.md [hunk 11]
```diff
@@ -1,21 +1,32 @@
...
+1:10273-104000
+1:105000-109000
+1:109000-112000
+```
+
+The chromosome names should correspond to the names found in the VCF files.
+
+## App Output
+
+The app returns VCF files in `*.vcf.gz` format, with corresponding tabix index files.
```
Comment (slzarate) [inline sequencing/vcf_slice/Readme.md]: Specify (`*.vcf.gz.tbi`)

File: sequencing/vcf_slice/Readme.md (hunk 8 above)
Comment (msimbirsky) [reply sequencing/vcf_slice/Readme.md:3]: Markdown ignores linebreaks, so this change wouldn't do anything.

File: sequencing/vcf_slice/dxapp.json (hunk 3 above)
Comment (msimbirsky) [reply sequencing/vcf_slice/dxapp.json]: A common filename for *.vcf.gz file is *.vcf.gz.tbi

**Comment by msimbirsky** [comment]
@slzarate @yihhwang Please see changes in response to your comments

**Pull request merged**